### ISSUE TITLE
[REQUEST]: LLVM optimiser pipeline

### ISSUE THREAD

**Issue by jgouly** [compiler-explorer/compiler-explorer#3832]
### Is your feature request related to a problem? Please describe

LLVM opt pipeline viewer was added for Clang in ed1a4e6b2794e096ecf049a64ed4944733f81b94, it would be great if this support could be extend to Rust.

### Describe the solution you'd like

`rustc` also supports passing arguments through to LLVM, so hopefully it can re-use the same infrastructure as was added for clang.

### Describe alternatives you've considered

You can pass `--emit llvm-ir` to rustc, but that only prints the final IR.

### Additional context

Not applicable

**Comment by jeremy-rifkin** [comment]
Yes this should definitely be doable, I was starting to look at this last night. `-C llvm-args="-print-after-all -print-before-all"` should work.
